crates/primitives/src/proof.rs

```diff
@@ -0,0 +1,42 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+
+use crate::{buf::Buf32, l1::L1BlockId, l2::L2BlockId};
+
+#[derive(Clone, Debug, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum RollupVerifyingKey {
+    // Verifying Key used to verify proof created using SP1
+    #[serde(rename = "sp1")]
+    SP1VerifyingKey(Buf32),
+    // Verifying Key used to verify proof created using Risc0
+    #[serde(rename = "risc0")]
+    Risc0VerifyingKey(Buf32),
+}
+
+/// `ProofKey` is an enumeration representing various identifiers for proofs.
+/// It is used to uniquely identify the proving tasks for different types of proofs.
+#[derive(
+    Debug, Clone, Copy, PartialEq, BorshSerialize, BorshDeserialize, Serialize, Deserialize,
+)]
+pub enum ProofKey {
```

Comment:
Do you think the term `ProofKey` will create ambiguity with the term ProvingKey ?

Reply:
Not so sure. 
But since we will be interacting with host (which internally has the ProvingKey), this should not create much ambiguity. AFAIK we haven't declared ProvingKey anywhere in our codebase.